src/main/java/org/folio/auth/authtokenmodule/AuthorizeApi.java

```diff
@@ -0,0 +1,724 @@
+      String address = ctx.request().remoteAddress().host();
+      String content = ctx.getBodyAsString();
+      JsonObject requestJson;
+      try {
+        requestJson = parseJsonObject(content, new String[] { "userId", "sub" });
+      } catch (Exception e) {
+        endText(ctx, 400, "Unable to parse content: ", e);
+        return;
+      }
+      String userId = requestJson.getString("userId");
+      String sub = requestJson.getString("sub");
+      String refreshToken = new RefreshToken(tenant, sub, userId, address).encodeAsJWE(tokenCreator);
+      JsonObject responseJson = new JsonObject().put("refreshToken", refreshToken);
+      endJson(ctx, 201, responseJson.encode());
+    } catch (Exception e) {
+      endText(ctx, 500, e);
+    }
+  }
+
+  /*
+   * Handle a request to sign a new token
+   * (Typically used by login module)
+   * Request content:
+   * {
+   * "payload" : { }
+   * }
+   */
+  private void handleSignToken(RoutingContext ctx) {
+    try {
+      logger.debug("Token signing request from {}", ctx.request().absoluteURI());
+      // tenant and okapiUrl are already checked in handleAuthorize
+      String tenant = ctx.request().headers().get(XOkapiHeaders.TENANT);
+      if (ctx.request().method() != HttpMethod.POST) {
+        endText(ctx, 400, "Unsupported operation: " + ctx.request().method().toString());
+        return;
+      }
+      final String postContent = ctx.getBodyAsString();
+      JsonObject json;
+      JsonObject payload;
+      try {
+        json = new JsonObject(postContent);
+      } catch (DecodeException dex) {
+        endText(ctx, 400, "Unable to decode '" + postContent + "' as valid JSON");
+        return;
+      }
+      payload = json.getJsonObject("payload");
+
+      if (payload == null) {
+        endText(ctx, 400, "Valid 'payload' field is required");
+        return;
+      }
+      logger.debug("Payload to create token from is {}", payload.encode());
+
+      if (!payload.containsKey("sub")) {
+        endText(ctx, 400, "Payload must contain a 'sub' field");
+        return;
+      }
+
+      String userId = payload.getString("user_id");
+      if (userId != null) {
+        permissionsSource.clearCacheUser(userId, tenant);
+      }
+      String username = payload.getString("sub");
+      Token token;
+      // auth 2.0 did not expose the "type" property which is now used internally.
+      // Only normal (access tokens) are exposed as well as dummy tokens (mod-users-bl).
+      if (payload.getBoolean("dummy", Boolean.FALSE)) {
+        token = new DummyToken(tenant, payload.getJsonArray("extra_permissions"), username);
+      } else {
+        token = new AccessToken(tenant, username, userId);
+      }
+      JsonObject responseObject = new JsonObject().put("token", token.encodeAsJWT(tokenCreator));
+      endJson(ctx, 201, responseObject.encode());
+    } catch (Exception e) {
+      endText(ctx, 400, e);
+    }
+  }
+
+  private void handleAuthorize(RoutingContext ctx) {
```

Comment:
I have difficulties to understand this method because it is very long, more than 300 lines. Can you move parts of it into sub-methods?

```
  someSubMethod();
  if (ctx.response().ended()) {
    return;
  }
```

Reply:
This will also make it easier to test some error handling code to increase code coverage: https://sonarcloud.io/component_measures?id=org.folio%3Amod-authtoken&metric=new_coverage&pullRequest=113&selected=org.folio%3Amod-authtoken%3Asrc%2Fmain%2Fjava%2Forg%2Ffolio%2Fauth%2Fauthtokenmodule%2FAuthorizeApi.java